paths_cli/parsing/tools.py

```diff
@@ -0,0 +1,36 @@
+import numpy as np
+
+def custom_eval(obj, named_objs=None):
+    string = str(obj)
+    # TODO: check that the only attribute access comes from a whitelist
+    namespace = {
+        'np': __import__('numpy'),
+        'math': __import__('math'),
+    }
```

Comment:
Added docstring:

    This allows certain user input to be treated as a simplified subset of
    Python. In particular, this is intended to allow simple arithmetic. It
    allows use of the modules numpy (as ``np``) and math, which provide
    potentially useful functions (e.g., ``cos``) as well as constants (e.g.,
    ``pi``).

My standard example here is when dealing with states defined by angles. MDTraj uses radians, but maybe the edge of your state is at 100 degrees. The user could calculate (or have Google calculate) "100 degrees in radians" and paste in the right answer, or they could just directly write:

```
100 * np.pi / 180
```

and the OPS CLI does the math for them.

I don't see a similar use case for adding MDTraj to the namespace, but if you have one, let me know. I could imagine adding unit support, but I don't want to do that until there's a specific case where it is needed.

Reply:
> I don't see a similar use case for adding MDTraj to the namespace, but if you have one, let me know. I could imagine adding unit support, but I don't want to do that until there's a specific case where it is needed.

Considering the usecase for this function I could not think of use case either. A possible addition to this function might be catching a `NameError` and stating what the current allowed modules are (with their namespace name)? feel free to leave it as is if you deem it out-of-scope for this PR

Reply:
> A possible addition to this function might be catching a `NameError` and stating what the current allowed modules are (with their namespace name)? feel free to leave it as is if you deem it out-of-scope for this PR

Good idea, but I think I'll leave that to the future. The specific message to report will probably depend on details that will only be available once we're parsing the AST of the user input. Reporting allowed modules will make more sense if the user asks for `foo.bar` rather than just `foo`. (Both create `NameError` on `foo`.) Parsing the AST will be needed to only allow a single level of attributes, or at least filter allowed multilevel attribute access (to forbid naughty tricks like `np.os.rmdir`).

The Wizard handles the error here: https://github.com/openpathsampling/openpathsampling-cli/blob/3d92cd248060aac02acc0e9686408ef914e2d62b/paths_cli/wizard/wizard.py#L122-L127

Example interaction:

```
🧙 What is the minimum allowed value for 'phi' in this volume? foo

👺 Sorry, I couldn't understand the input 'foo'
   Here's the error I got:
   NameError: name 'foo' is not defined

🧙 What is the minimum allowed value for 'phi' in this volume? 100 * np.pi / 180

🧙 What is the maximum allowed value for 'phi' in this volume?
```

When this is reused in #43, there will need to be some other error handling.